internal/tools/policyenginesimulator/README.md

```diff
@@ -0,0 +1,217 @@
+the policy. These commands take the form of JSON, *e.g.*:
+
+``` json
+[
+    {
+        "name": "load_fragment",
+        "input": {
+            "issuer": "did:web:contoso.github.io",
+            "feed": "contoso.azurecr.io/custom",
+            "namespace": "custom",
+            "local_path": "custom.rego"
+        }
+    },
+    {
+        "name": "mount_device",
+        "input": {
+            "target": "/mnt/layer0",
+            "deviceHash": "16b514057a06ad665f92c02863aca074fd5976c755d26bff16365299169e8415"
+        }
+    },
+    {
+        "name": "mount_overlay",
+        "input": {
+            "target": "/mnt/overlay0",
+            "containerID": "container0",
+            "layerPaths": [
+                "/mnt/layer0"
+            ]
+        }
+    },
+    {
+        "name": "create_container",
+        "input": {
+            "containerID": "container0",
+            "argList": [
+                "/pause"
+            ],
+            "envList": [
+                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+                "TERM=xterm"
+            ],
+            "mounts": [],
+            "workingDir": "/",
+            "sandboxDir": "/sandbox",
+            "hugePagesDir": "/hugepages"
+        }
+    }
+]
+```
+
+Each command has a name (corresponding to the API enforcement point) and then
+an input which will be passed directly to the policy.
+
+## Data
+
+If the authored policy requires certain values in the Rego data structure to
+function, or if the author wants to test enforcement behavior given a known
+starting state, they can provide an optional initial data file as an argument
+to the tool. This should take the form of a single JSON object, *e.g.*:
+
+``` json
+{
+    "defaultMounts": [/*some mount constraint objects*/],
+    "privilegedMounts": [/*some mount constraint objects*/],
+    "sandboxPrefix": "sandbox://",
+    "hugePagesPrefix": "hugepages://"
+}
+```
+
+## Log
+
+To aid in debugging, the user can specify a log file. This will enable logging
+at the `Info` level, which consists of the output of any Rego `print()` calls
+from the policy.
+
+## Log Level
+
+There are several log levels that the user can use to gain greater insight into
+policy enforcement:
+
+|    Name    |                       Description                         |
+| ---------- | --------------------------------------------------------- |
+| `None`     | Used when no log file is provided via the `-log` argument |
+| `Info`     | Outputs the results of Rego `print()` statements.         |
+| `Results`  | Outputs the results returned by each policy query         |
+| `Metadata` | Outputs the entire metadata state after each query        |
+
+## Policy
+
+This can be any Rego with a package name of `policy`, though policies which do
+not define the required enforcement point rules will result in enforcement
+failures. Here are some example policies:
```

Comment:
Since the examples you have in the samples directory are more detailed, I would just note to see them for more extensive examples.

Reply:
Agreed, link added.